internal/codegen/tpl_stencil.go

```diff
@@ -47,6 +49,26 @@ func (s *TplStencil) GetModuleHook(name string) []interface{} {
 	k := s.s.sharedData.key(s.t.Module.Name, name)
 	v := s.s.sharedData.moduleHooks[k]
 
+	// sort the output so that it's deterministic
+	sort.Slice(v, func(i, j int) bool {
+		// We have to hash the values because they're not the same
+		// type, so we can't compare them directly
+		hi, herr := hashstructure.Hash(v[i], hashstructure.FormatV2, nil)
+		hj, jerr := hashstructure.Hash(v[j], hashstructure.FormatV2, nil)
```

Comment:
you might want to pre-hash all entries into module hooks so that you're not re-hashing the same entries